pkg/azuredisk/controllerserver.go

```diff
@@ -284,8 +284,17 @@ func (d *Driver) ControllerPublishVolume(ctx context.Context, req *csi.Controlle
 	if len(nodeID) == 0 {
 		return nil, status.Error(codes.InvalidArgument, "Node ID not provided")
 	}
-	nodeName := types.NodeName(nodeID)
 
+	nodeInfo, err := d.NodeGetInfo(ctx, &csi.NodeGetInfoRequest{})
+	if err != nil {
+		return nil, err
+	}
+
+	if nodeInfo.NodeId != nodeID {
```

Comment:
`nodeInfo.NodeId` is the nodeid of this driver is running:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/b6f67023443d568b97b5b4fb9b7272d6b5f32530/pkg/azuredisk/nodeserver.go#L262-L263

while `nodeID` is the target VM which is going to attach a disk, I think it's different.

Reply:
Okay 😅